plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableSearchQuery.php

```diff
@@ -119,7 +119,29 @@ private function generate_join_for_search_filter( $search_filter ) : string {
 				LEFT JOIN $items_table AS search_query_items ON search_query_items.order_id = $orders_table.id
 			";
 		}
-		return '';
+
+		/**
+		 * Filter to support adding a new search filter.
+		 * Provide a JOIN clause for a new search filter. This should be used along with `woocommerce_hpos_admin_search_filters`
+		 * to declare a new custom filter, and `woocommerce_hpos_generate_where_for_search_filter` to generate the WHERE
+		 * clause.
+		 *
+		 * Hardcoded JOINS (products) cannot be modified using this filter for consistency.
+		 *
+		 * @since 8.9.0
+		 *
+		 * @param string $join The JOIN clause.
+		 * @param string $search_term The search term.
+		 * @param string $search_filter The search filter. Use this to bail early if this is not filter you are interested in.
+		 * @param OrdersTableQuery $query The order query object.
+		 */
+		return apply_filters(
```

Comment:
Hmm, I am not sure if we can really validate a query short of actually running it. For example, it's allowed to return empty string, but also INNER JOIN can be written without using the JOIN keyword (although I am not really expecting this to be done here). Only thing we can validate is that the return is string, which is already being done by the presence of the return type in the function definition. So maybe this is good enough as is, wdyt?

Reply:
> Only thing we can validate is that the return is string, which is already being done by the presence of the return type in the function definition.

Ah, I missed that. Yeah, this is probably good enough, since at least it will throw an error if a filter injects a non-string value.